### DISCUSSION TITLE
Build Error for Android

### DISCUSSION THREAD

**Discussion by Philnue** [flutter-stripe/flutter_stripe#1675] [Q&A]
**Describe the bug**
I am getting an error for the adaption for the themes when building

**Expected behavior**
 
`plugins {
    id "com.android.application"
    id "kotlin-android"
    id "dev.flutter.flutter-gradle-plugin"
}

def localProperties = new Properties()
def localPropertiesFile = rootProject.file('local.properties')
if (localPropertiesFile.exists()) {
    localPropertiesFile.withReader('UTF-8') { reader ->
        localProperties.load(reader)
    }
}

def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
if (flutterVersionCode == null) {
    flutterVersionCode = '1'
}

def flutterVersionName = localProperties.getProperty('flutter.versionName')
if (flutterVersionName == null) {
    flutterVersionName = '1.0'
}

android {
    namespace "com.x.x"
    compileSdkVersion 31
    ndkVersion flutter.ndkVersion

   compileOptions {
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }

    kotlinOptions {
        jvmTarget = '11'
    }

    sourceSets {
        main.java.srcDirs += 'src/main/kotlin'
    }

    defaultConfig {
        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
        applicationId "com.x.x"
        // You can update the following values to match your application needs.
        // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-gradle-build-configuration.
        minSdkVersion 21
        targetSdkVersion 31
        versionCode flutterVersionCode.toInteger()
        versionName flutterVersionName
    }

    buildTypes {
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig signingConfigs.debug
        }
    }
}

flutter {
    source '../..'
}

dependencies {
   // implementation 'com.google.android.material:material:<version>'
}
`

This is my app/build.gradle

`buildscript {
    ext.kotlin_version = '1.8.10'
    repositories {
        google()
        mavenCentral()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:8.3.0'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
    }
}
allprojects {
    repositories {
        google()
        mavenCentral()
    }
}

rootProject.buildDir = '../build'
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"
}
subprojects {
    project.evaluationDependsOn(':app')
}

tasks.register("clean", Delete) {
    delete rootProject.buildDir
}`

THis is the project build.gradle file

And copied the themes from guide

* What went wrong:
Execution failed for task ':app:processReleaseResources'.
> A failure occurred while executing com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask$TaskAction
   > Android resource linking failed
     ERROR: AAPT: error: resource style/Theme.AppCompat.Light.NoActionBar (aka com.x.x:style/Theme.AppCompat.Light.NoActionBar) not found.
     error: resource style/Theme.AppCompat.DayNight.NoActionBar (aka com.x.x:style/Theme.AppCompat.DayNight.NoActionBar) not found.
     error: resource style/Theme.MaterialComponents (aka com.x.x:style/Theme.MaterialComponents) not found.
     error: resource style/Theme.MaterialComponents (aka com.x.x:style/Theme.MaterialComponents) not found.
     error: failed linking references.

**Comment by lpylpyleo** [comment]
Same problem. 
Specifying `flutter_stripe: 10.0.0` doesn't work. I have to use `dependency_overrides`

```yaml
dependencies:
  flutter_stripe: 10.0.0

dependency_overrides:
  stripe_platform_interface: 10.0.0
  stripe_android: 10.0.0
  stripe_ios: 10.0.0
```

**Comment by Philnue** [comment]
Then It was working ?
You had the same error

**Comment by Philnue** [comment]
@lpylpyleo  can you please share your theme.xml, manifest, and gradle files ? For me it is not working

**Comment by Migo-17** [comment]
@Philnue can you try setting the gradle version to 8.1.4                
    -classpath 'com.android.tools.build:gradle:8.1.4'
and add this in pubspec.yaml 
dependency_overrides:
  stripe_android: 10.0.0

**Comment by Philnue** [comment]
Only for Android ? I try but I think I tried all variants of changes. @Migo-17. I alsocreated a new project only with stripe not working. Most of the Time I get. The version of Metadata is 1.9.0 but expected version is 1.7.1 and I dont know how to fix it

**Comment by lpylpyleo** [comment]
@Philnue Try running `flutter clean; flutter pub get` and build again. It works for me.

**Comment by Philnue** [comment]
@lpylpyleo  I already did, I am testing an older Git Commit which was working correctly and testing it. The funny thing is the older one is working the latest with the same chagnes not working. I compared everything. So I am gonna go with the oldest and make the changes there. THe older commit is using 9.5.1+1 Version of Flutter Stripe

**Comment by Migo-17** [comment]
> Only for Android ? I try but I think I tried all variants of changes. @Migo-17. I alsocreated a new project only with stripe not working. Most of the Time I get. The version of Metadata is 1.9.0 but expected version is 1.7.1 and I dont know how to fix it

yes, only changing the android version fixed it for me

**Comment by Philnue** [comment]
@Migo-17  how exactly can you change the android version ? The compile sdk

**Reply by Migo-17** [reply]
In the dependencies in build.gradle inside the android main folder I changed it to   
    dependencies {
        classpath 'com.android.tools.build:gradle:8.1.4'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"  
    }

**Comment by remonh87** [comment]
I would recommend looking at the troubleshoot page and check if all requirements for android are met: https://docs.page/flutter-stripe/flutter_stripe/troubleshooting

**Reply by Philnue** [reply]
I did I although tried the example which didn't work

**Reply by imasif** [reply]
same here
